src/groups/orthogonal_group.jl

```diff
@@ -0,0 +1,653 @@
+    e::Identity{MatrixMultiplicationGroupOperation},
+    g,
+)
+    Y = ManifoldsBase.allocate_result(G, log, g)
+    log!(G, Y, e, g)
+    return Y
+end
+# Resolve an ambiguity compared to the general matrix multiplication definitions
+# TODO: Maybe combine the following 3 dispatches?
+function Base.log(
+    G::CommonUnitarySubGroups{ManifoldsBase.ℝ,ManifoldsBase.TypeParameter{Tuple{2}}},
+    e::Identity{MatrixMultiplicationGroupOperation},
+    ::Identity{MatrixMultiplicationGroupOperation},
+)
+    return zero_vector(G, e)
+end
+function Base.log(
+    G::CommonUnitarySubGroups{ManifoldsBase.ℝ,ManifoldsBase.TypeParameter{Tuple{3}}},
+    e::Identity{MatrixMultiplicationGroupOperation},
+    ::Identity{MatrixMultiplicationGroupOperation},
+)
+    return zero_vector(G, e)
+end
+function Base.log(
+    G::CommonUnitarySubGroups{ManifoldsBase.ℝ,ManifoldsBase.TypeParameter{Tuple{4}}},
+    e::Identity{MatrixMultiplicationGroupOperation},
+    ::Identity{MatrixMultiplicationGroupOperation},
+)
+    return zero_vector(G, e)
+end
+
+function ManifoldsBase.log!(
+    ::CommonUnitarySubGroups{ManifoldsBase.ℝ,ManifoldsBase.TypeParameter{Tuple{2}}},
+    X,
+    ::Identity{MatrixMultiplicationGroupOperation},
+    g,
+)
+    @assert size(X) == (2, 2)
+    @assert size(g) == (2, 2)
+    @inbounds α = atan(g[2, 1], g[1, 1])
+    @inbounds begin
+        X[1, 1] = 0
+        X[2, 1] = α
+        X[1, 2] = -α
+        X[2, 2] = 0
+    end
+    return X
+end
+function ManifoldsBase.log!(
+    G::CommonUnitarySubGroups{ℝ,ManifoldsBase.TypeParameter{Tuple{3}}},
+    X::AbstractMatrix,
+    e::Identity{MatrixMultiplicationGroupOperation},
+    q::AbstractMatrix,
+)
+    cosθ = (tr(q) - 1) / 2
+    if cosθ ≈ -1
+        eig = eigen_safe(q)
+        ival = findfirst(λ -> isapprox(λ, 1), eig.values)
+        inds = SVector{3}(1:3)
+        ax = eig.vectors[inds, ival]
+        return get_vector!(G, X, e, π * ax, LieAlgebraOrthogonalBasis())
+    end
+    X .= q ./ usinc_from_cos(cosθ)
+    # project onto 𝔰𝔬(3) for numerical stability
+    return project!(G, X, e, X)
+end
+function ManifoldsBase.log!(
+    G::CommonUnitarySubGroups{ℝ,ManifoldsBase.TypeParameter{Tuple{4}}},
+    X::AbstractMatrix,
+    e::Identity{MatrixMultiplicationGroupOperation},
+    q::AbstractMatrix,
+)
+    cosα, cosβ = cos_angles_4d_rotation_matrix(q)
+    α = acos(clamp(cosα, -1, 1))
+    β = acos(clamp(cosβ, -1, 1))
+    if α ≈ 0 && β ≈ π
+        A² = Symmetric((q - I) ./ 2)
+        P = eigvecs(A²)
+        E = similar(q)
+        fill!(E, 0)
+        @inbounds begin
+            E[2, 1] = -β
+            E[1, 2] = β
+        end
+        copyto!(X, P * E * transpose(P))
+    else
+        det(q) < 0 && throw(
+            DomainError(
+                "The Lie group logarithm is not defined for $q with a negative determinant ($(det(q)) < 0). Point `q` is in a different connected component of the manifold $G",
+            ),
+        )
+        copyto!(X, real.(log(q)))
```

Comment:
@Mateusz can you maybe take a Loog at the current error in one of the corner cases? I noticed I dod not take over `log_safe!`, maybe with the goal of simplifying the code. It seems to be the only difference of the 4x4 log compared to the Manifolds.jl implementation. I pent the whole morning trying to fix this but even _if_ I now copy over the (super long and techical) `log_safe!` cf https://github.com/JuliaManifolds/Manifolds.jl/blob/b52472bc116987e93b10f58b5e0599db4534728e/src/utils.jl#L83. Nothing works. I spent the whole morning trying to fix this. 
Sadly this code is more clever than I am, so I have no clue how to fix the one corner case failing.

Reply:
Sure, I will check it.

Reply:
Fixed :slightly_smiling_face:

Reply:
I would literally never have found that. Maybe I'll leave these technical special cases to others in the future then.
Thanks for fixing it. I actually removed a few get_vector get_coordinate calls in transferring the code, since it was sometimes cleaner afterwards.
I feel the same is the case here as well.

Reply:
Hm, on CI exactly the same corner case (the third) is failing with the same error as yesterday.

Reply:
Hm, weird, I think it worked for me locally but now when I try again, I get the same failure as on CI. Curiously, Manifolds.jl has the exact same issue when I modify the test to match your in LieGroups.jl. Numerical errors are enough to make it work :thinking:

Reply:
I copied the tests over – especially the boundary cases, so I do not understand “when you modify the test to match”. I also do not understand the boundary cases not what you refer to with “numerical errors make it work”.

are the corner cases maybe wrong?

Reply:
You switched `exp(X)` with `exp_lie(M, X)`. They are off by `eps()` in two places, which is enough to change the outcome but not enough to fail Manifolds.jl tests. I'm investigating.

Reply:
Well, we do not _have_ `exp_lie` here with, so I can not do magic? I could do `exp(G, Identity(G), X)` which is its replacement, but then we compare A to A, which is equal? Well,...I was lost on this a few days ago already, that did not yet improve. But I also get too easily lost on those super-technical details. Will better not try to implement those again ;)

 I am looking forward to learning what the problem here then actually is.

Reply:
I've looked at Gallier's paper about these computations. It looks like original implementation hoped that Julia's matrix `log` handles everything correctly except the case `α ≈ 0 && β ≈ π`. It clearly does not, and the failing test demonstrates the problem. I suggest marking this case as broken. I'm sure it could be fixed based on that paper but it looks like a work for a day or two.

Reply:
Ok, that sounds fair. Actually just exp/log worked often fine before (when I tested the code here that is).
I think the case failing here is `α ≈ π/2 && β ≈ π` ? but sure, until someone (probably you) finds time for fixing that case, let's leave this one marked as broken.

Reply:
Yes, that's exactly the failing case.